src/models/metadata.ts

```diff
@@ -13,6 +13,30 @@ import { Timestamp } from './timestamp';
 
 type MetadataType = Root | Timestamp | Snapshot | Targets;
 
+/***
+ * A container for signed TUF metadata.
+ *
+ * Provides methods to convert to and from dictionary, read and write to and
+ * from file and to create and verify metadata signatures.
+ *
+ * ``Metadata[T]`` is a generic container type where T can be any one type of
+ * [``Root``, ``Timestamp``, ``Snapshot``, ``Targets``]. The purpose of this
+ * is to allow static type checking of the signed attribute in code using
+ * Metadata::
+ *
+ * root_md = Metadata[Root].from_file("root.json")
+ * # root_md type is now Metadata[Root]. This means signed and its
+ * # attributes like consistent_snapshot are now statically typed and the
+ * # types can be verified by static type checkers and shown by IDEs
+ *
+ * Using a type constraint is not required but not doing so means T is not a
+ * specific type so static typing cannot happen. Note that the type constraint
+ * ``[Root]`` is not validated at runtime (as pure annotations are not available
+ * then).
+ *
+ * Apart from ``expires`` all of the arguments to the inner constructors have
+ * reasonable default values for new metadata.
+ */
```

Comment:
Does this map to the JS implementation? Should `from_file` to `fromJSON` for example?